srv/salt/ceph/monitoring/grafana/init.sls

```diff
@@ -23,6 +25,26 @@ grafana-dashboards-ceph:
         enabled = true
         path = /var/lib/grafana-dashboards-ceph
 
+wait-for-grafana-http:
+  cmd.run:
+    - require:
+      - service: grafana-server
+    - name: |
+         i=0
+         curl -s -H "Content-Type: application/json" -XGET http://admin:admin@localhost:3000/api/datasources
+         curl_ret=$?
+         until [ $curl_ret -eq 0 ] || [ $i -eq 24 ]; do
+           sleep 5
+           curl -s -H "Content-Type: application/json" -XGET http://admin:admin@localhost:3000/api/datasources
+           curl_ret=$?
+           i=$((i + 1))
+         done;
+         if [ $curl_ret -ne 0 ] ; then
+             echo "ERROR: The grafana http server faild to start after $((i * 5))s"
```

Comment:
There's probably an off-by-one error in there -> the conditional should perhaps be `-le` instead of `-lt` to ensure that the number of seconds reported in the error is accurate (not off by 5 seconds).